lib/cf_obs_binary_builder/templates/buildpack-go-based.spec.erb

```diff
@@ -42,22 +42,26 @@ BuildRoot:      %{_tmppath}/%{name}-%{version}-build
 %build
 source ./.envrc
 
-if [ -e go.mod ]; then
-
+if [ -f go.mod ]; then
   # Starting with Go 1.11 modules
-  # installs buildpack-packager from generated vendor
-  go install -mod=vendor buildpack-packager.go
+  # installs buildpack-packager from generated
+  mkdir /tmp/bin
+  export PATH=$PATH:/tmp/bin
+  mv buildpack-packager /tmp/bin/buildpack-packager
+  chmod +x /tmp/bin/buildpack-packager
 
-  # With go modules, we are building up the vendor/ folder before landing in obs
-  # both for the buildpack-packager and also for the buildpack itself.
-  VENDOR_GOPATH=%{_builddir}/go
-  export GO111MODULE=off # Disable new behavior, needs network access
-  rm -rf go.mod buildpack-packager.go
-  export GOPATH=$VENDOR_GOPATH:$GOPATH
-  mkdir -p $VENDOR_GOPATH/src/
-  mv vendor/* $VENDOR_GOPATH/src/
-  mkdir -p $VENDOR_GOPATH/src/github.com/cloudfoundry/<%= name %>-buildpack
-  cp -rf src/ $VENDOR_GOPATH/src/github.com/cloudfoundry/<%= name %>-buildpack
+  if [ ! -f "vendor/modules.txt" ]; then
+    # With go modules, we are building up the vendor/ folder before landing in obs
+    # both for the buildpack-packager and also for the buildpack itself.
+    VENDOR_GOPATH=%{_builddir}/go
+    export GO111MODULE=off # Disable new behavior, needs network access
```

Comment:
WIth go.mod deleted this should be needed I think, the non-modules behaviour will be picked automatically but I'm not sure.